File: android/src/main/java/de/schindlergmbh/plugins/capacitor/c6gunapi/C9GunApiCapacitorPluginPlugin.java
package de.schindlergmbh.plugins.capacitor.c6gunapi;

import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

@CapacitorPlugin(name = "C9GunApiCapacitorPlugin")
public class C9GunApiCapacitorPluginPlugin extends Plugin {

    private C9GunApiCapacitorPlugin implementation = new C9GunApiCapacitorPlugin();

    @PluginMethod
    public void echo(PluginCall call) {
        String value = call.getString("value");

        JSObject ret = new JSObject();
        ret.put("value", implementation.echo(value));
        call.resolve(ret);
    }
}

File: ios/Plugin/C9GunApiCapacitorPluginPlugin.swift
import Foundation
import Capacitor

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(C9GunApiCapacitorPluginPlugin)
public class C9GunApiCapacitorPluginPlugin: CAPPlugin {
    private let implementation = C9GunApiCapacitorPlugin()

    @objc func echo(_ call: CAPPluginCall) {
        let value = call.getString("value") ?? ""
        call.resolve([
            "value": implementation.echo(value)
        ])
    }
}
